solutions/devsprint-matheus-reis-7/FinanceApp/Modules/ActivityDetails/ActivityDetailsView.swift

```diff
@@ -89,10 +81,38 @@ class ActivityDetailsView: UIView {
         return button
     }()
 
-
+    // MARK: - Initializer
+    // TODO: Implement viewCodable procotol
     init() {
         super.init(frame: .zero)
+        setup()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Protocol conformance
+    // TODO: Create a ActivityDetailsViewModel just for `price: String`?
+    func show(viewModel: ActivityDetails) {
+        activityNameLabel.text = viewModel.name
+        categoryLabel.text = viewModel.category
+        // TODO: Method overloading? Unecessary computation? Float -> String
+        priceLabel.text = String(viewModel.price)
+        timeLabel.text = viewModel.time
+    }
 
+    @objc
+    func reportButtonPressed() {
+
+        delegate?.didPressReportButton()
+    }
+
+}
+
+// TODO: Implement viewCodable procotol
+private extension ActivityDetailsView {
+    private func setup() {
```

Comment:
As you already using a private extension, you don't have necessity of using private functions :) 
```suggestion
    func setup() {
```